osrf/vrx
# [Garden] Port scan, dock, deliver - Part 1

This pull request partially ports the `ScanDockScoringPlugin` from Gazebo classic. In particular, it ports the functionality related with the color sequence checker.

### How to test it?

* Launch the scan_dock_deliver_task world:
```
ros2 launch vrx_gz competition.launch.py world:=scan_dock_deliver_task
```

* On a new terminal check the `/vrx/task/info` topic:
```
ros2 topic  echo /vrx/task/info
```

* Wait until we enter into `running` state (around 20 secs), and then, submit a correct color sequence. On a new terminal launch:
```
ros2 topic pub /vrx/scan_dock_deliver/color_sequence --once ros_gz_interfaces/msg/StringVec "header:
  stamp:
    sec: 0
    nanosec: 0
  frame_id: ''
data: ["red", "green", "blue"]"
```
Verify in your previous terminal that you get 10 points.

* Additionally, you can rerun the simulation and submit an incorrect sequence. You shouldn't get any points. Additional submissions should be ignored, as there's only one attempt.
